kbase-extension/static/kbase/js/util/jobLogViewer.js

```diff
@@ -998,53 +1163,73 @@ define([
                 },
                 handle: handleJobDoesNotExistUpdate,
             });
-            listeners.push(ev);
         }
 
-        function stopEventListeners() {
-            runtime.bus().removeListeners(listeners);
+        function stopEventListeners(listeners = []) {
+            if (!listeners.length) {
+                listeners = Object.keys(listenersByType);
+            }
+            bus.removeListeners(listeners.map((l) => listenersByType[l]));
+        }
+
+        function renderWidgetState() {
+            if (developerMode) {
+                ui.setContent(
+                    'dev.widget-state',
+                    JSON.stringify(
+                        {
+                            fsm: fsm.getCurrentState().state,
+                            looping: looping,
+                            awaitingLog: awaitingLog,
+                            listeningForJob: listeningForJob,
+                        },
+                        null,
+                        1
+                    )
+                );
+            }
         }
 
         // LIFECYCLE API
         function renderFSM() {
             const state = fsm.getCurrentState();
-
+            renderWidgetState();
             // Button state
             if (state.ui.buttons) {
                 state.ui.buttons.enabled.forEach((_button) => {
-                    ui.enableButton(_button);
+                    try {
+                        ui.enableButton(_button);
+                    } catch (err) {
+                        // eslint-disable-next-line no-empty
+                    }
                 });
                 state.ui.buttons.disabled.forEach((_button) => {
-                    ui.disableButton(_button);
-                });
-            }
-
-            // Element state
-            if (state.ui.elements) {
-                state.ui.elements.show.forEach((element) => {
-                    ui.showElement(element);
-                });
-                state.ui.elements.hide.forEach((element) => {
-                    ui.hideElement(element);
+                    try {
+                        ui.disableButton(_button);
+                    } catch (err) {
+                        // eslint-disable-next-line no-empty
+                    }
```

Comment:
in some states, the buttons have been removed altogether -- this just stops the UI from throwing errors if that is the case.